### DISCUSSION TITLE
Using openrailway.org as tiles?

### DISCUSSION THREAD

**Discussion by oyvindsolstad** [javalent/obsidian-leaflet#256] [Q&A]
The read me says: 

> "Real world maps are created if the image parameter is not provided. These maps by default will load the OpenStreetMap map, but additional tile servers can be provided using the tileServer parameter."

How? Can I use https://www.openrailwaymap.org as tiles? If so, what would the code be?

**Comment by valentine195** [comment]
According to [this page](https://wiki.openstreetmap.org/wiki/OpenRailwayMap/API#Usage_in_Leaflet), their tile server is this:

`http://{s}.tiles.openrailwaymap.org/standard/{z}/{x}/{y}.png`

So what you can do is:

````
```leaflet
id: rail
osmLayer: false
tileServer: http://{s}.tiles.openrailwaymap.org/standard/{z}/{x}/{y}.png
```
````

However, it looks like this supposed to be a overlay, not a unique layer. Let me think about a way to add this feature in.

**Reply by oyvindsolstad** [reply]
Thanks! Yes, I see that the rest of the information on the map is gone, it's "just" the rails now. But that alone helps me a lot!

Example: One of the stations on the Bergensbanen line. All the tracks clearly marked. 

![Skjermbilde 2021-12-10 kl  17 21 17](https://user-images.githubusercontent.com/4631971/145606843-6c62e2af-456e-45ab-ac5a-00311b9e0767.png)

**Reply by oyvindsolstad** [reply]
Now I don't have to chase Geojson-files of all tracks (I work in Vy - a railway company). They are just there, highly visible. In this screenshot, the orange markers are stations I've added to Obsidian.

![Skjermbilde 2021-12-10 kl  17 29 41](https://user-images.githubusercontent.com/4631971/145608228-dc434660-df30-4196-b1be-06d1baf056f2.png)

**Reply by valentine195** [reply]
Hey, I just pushed a new release. Once you can update to it, you can now do this:

````
```leaflet
id: test 3
height: 100%
tileOverlay: http://{s}.tiles.openrailwaymap.org/standard/{z}/{x}/{y}.png|Rails
```
````
![Zok8QmdmeV](https://user-images.githubusercontent.com/38669521/145615261-0e11e143-7ace-44a3-a165-cadfce438fa2.gif)

**Reply by valentine195** [reply]
See [here](https://github.com/valentine195/obsidian-leaflet-plugin#tile-servers-and-tile-overlays).

**Reply by oyvindsolstad** [reply]
Wooow! Super!!

**Comment by oyvindsolstad** [comment]
Found this site, where you can see all the different tileservers supported. Experimenting ahead!!

https://leaflet-extras.github.io/leaflet-providers/preview/

**Comment by oyvindsolstad** [comment]
I've been playing around with the different services found on the [Leaflet provider page](https://leaflet-extras.github.io/leaflet-providers/preview/). Some of them are super interesting. To name a few: 

• TomTom maps (imho a far better traffic service than Apple and Google Maps)
• openAIP - show airports, zones where you can't fly drones etc. 
• OpenWeather - possible to show live weather on maps

So I tried to get them to work on the maps. Any suggestions on how? 

An example with OpenWeather. I've signed up for a free account and got an API key.

Code is:

````leaflet
id: AustinTestMAP
lat: 31.039800374465774
long:  -97.71451338616012
tileOverlay:
  - http://{s}.tiles.openrailwaymap.org/standard/{z}/{x}/{y}.png|Rails
  - http://{s}.tiles.openrailwaymap.org/signals/{z}/{x}/{y}.png|Signals
  - http://{s}.tiles.openrailwaymap.org/maxspeed/{z}/{x}/{y}.png|Speed
  - http://{s}.tile.openweathermap.org/map/clouds/{z}/{x}/{y}.png?appid={nnnnnnnnnn}|Weather
height: 500px
width: 100%
minZoom: 3
maxZoom: 20
defaultZoom: 15
unit: meters
scale: 1
darkMode: false
````

Where appid says nnnnnnnn, I have used my proper API key. But it doesn't work. The rails show up correctly and by turning on and off layers, I can see both, rails, signals and speed, as described [here](https://wiki.openstreetmap.org/wiki/OpenRailwayMap/API#Tiles). But the weather is missing. 

---

So that's one thing I'm curios about. The other is how to add API keys for the other services in the list?

**Reply by valentine195** [reply]
Let me investigate, but right now the plugin only supports tile servers than can be entered as a url. If they need additional information, they can’t be used. I could possibly enable this with a YAML-object as the tile server parameter though, but you’d have to be comfortable putting your login/API info in the note text.

**Reply by oyvindsolstad** [reply]
No worries, I can just use the ones where you put everything on a line. 

Did I you mean putting the api key in YAML? That would work for me.
